src/storage.rs

```diff
@@ -68,41 +71,73 @@ impl Default for AccountBasic {
     }
 }
 
-/// EVM Code, currently mapping to REVM's [ByteCode::LegacyAnalyzed].
-// TODO: Support raw legacy & EOF
-#[derive(Debug, Clone, PartialEq, Default, Serialize, Deserialize)]
-pub struct EvmCode {
-    /// Bytecode with 32 zero bytes padding
+/// Analyzed legacy code.
+// TODO: Store unpadded bytecode and pad on revm conversion.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct LegacyCode {
+    /// Bytecode with 32 zero bytes padding.
     bytecode: Bytes,
-    /// Original bytes length
+    /// Original bytes length.
     original_len: usize,
     /// Jump table.
     jump_table: Arc<BitVec<u8>>,
 }
 
+/// EIP7702 delegated code.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Eip7702Code {
+    /// Address of the EOA which will inherit the bytecode.
+    delegated_address: Address,
+    /// Version of the bytecode.
+    version: u8,
+}
+
+/// EVM Code, currently mapping to REVM's [ByteCode].
+// TODO: Support EOF
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum EvmCode {
+    /// Maps both analyzed and non-analyzed REVM legacy bytecode.
+    Legacy(LegacyCode),
+    /// Maps delegated EIP7702 bytecode.
+    Eip7702(Eip7702Code),
+}
+
 impl From<EvmCode> for Bytecode {
     fn from(code: EvmCode) -> Self {
         // TODO: Better error handling.
-        // A common trap would be converting a default [EvmCode] into
```

Comment:
Rm'd this comment as `default` is no longer derived.

Reply:
Nice!